pkg/kubelet/cri/remote/conversion_test.go

```diff
@@ -1,409 +0,0 @@
-/*
-Copyright 2021 The Kubernetes Authors.
-
-Licensed under the Apache License, Version 2.0 (the "License");
-you may not use this file except in compliance with the License.
-You may obtain a copy of the License at
-
-    http://www.apache.org/licenses/LICENSE-2.0
-
-Unless required by applicable law or agreed to in writing, software
-distributed under the License is distributed on an "AS IS" BASIS,
-WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
-See the License for the specific language governing permissions and
-limitations under the License.
-*/
```

Comment:
@mrunalp was the plan to remove `v1alpha2` support before diverging both CRI versions? If so, have we already agreed on a version when to remove `v1alpha2`?

Reply:
@saschagrunert taking into account what you said in https://github.com/kubernetes/kubernetes/pull/110618#issuecomment-1157551147, it might be then a good interim solution to update both (v1 and alpha) for now as @giuseppe already did?

The alpha interface will be deprecated before this feature gate is enabled by default, if we remove alpha in October/November, so it is something mostly no one will notice. Does it sound ok? Or too hacky?

Reply:
Awesome, thanks!

Reply:
Speaking for the container runtime, I don't think we should force old releases of container runtimes to add new features. For example, containerd has a rule to not add new features to service branches. I believe kubernetes has the same type of rule for it's APIs. If the change is "non-breaking" that may be an ok change such as how we did the optional changes last release. I expected to see no more changes to v1alpha2, and after n-2 releases of k8s having added v1, I expected v1alpha2 to drop from kubelet and the container runtimes.

Reply:
@mikebrow I think we weren't clear, we weren't trying to force old container runtimes releases to implement this, rather this was a way out to not break the current test that assert they v1 and alpha are equal, until we can remove alpha in a few releases. The expectation is that only v1 runtimes will implement the feature (the userns feature gate will still be alpha when the CRI alpha interface is dropped, so probably no one will even notice and no changes in the runtimes that only implement the alpha CRI is needed).

We are fine either way: making the interfaces differ now or adding the change for both, but we need a way forward to merge this changes :)

Reply:
nod.. container runtimes will just have to clarify support of user ns option in "optional" fields .. my concern, per chat, is more around new apis than optional extensions